app/ts/Legend/LegendWrapper/LegendUtil.ts

```diff
@@ -0,0 +1,24 @@
+import IAlternative from '@shared/interface/IAlternative';
+import _ from 'lodash';
+
+export function initLegend(
+  legend: Record<string, string>,
+  alternatives: IAlternative[]
+): Record<string, string> {
+  return legend
+    ? legend
+    : _(alternatives).keyBy('id').mapValues('title').value();
+}
+
+export function generateSingleLetterLegend(
+  alternatives: IAlternative[]
+): Record<string, string> {
+  return _.fromPairs(
+    _.zip(
```

Comment:
nice :)

The `zip` isn't stricty necessary,
```
_.map(alternatives, (alternative, index) => [alternative.id, String.fromCharCode(65+index)])
```
to generate the pairs should also work, and is a bit simpler